### ISSUE TITLE
Possible oversight in with-odd-context macro

### ISSUE THREAD

**Issue by scymtym** [guicho271828/trivialib.bdd#1]
The default form of the ``operation`` parameter of the ``with-odd-context`` macro is ``#'bdd-apply``.

When evaluated during macroexpansion, this form evaluates to the function object to which the name ``bdd-apply`` is fbound. Unless a different value is supplied for the parameter, this function object is spliced into the expansion.

Such a use of the macro in a file compilation context leads to an attempt to dump the function object into the fasl file as a literal.

This behavior seems unintentional and like an oversight.

The test suite did not reveal this problem because fiveam currently evaluates test bodies without compiling them. See sionescu/fiveam#40 for context.

**Comment by guicho271828** [comment]
Thanks for the report. Do you have plan to use this library like real soon? I would recommend CL-CUDD if you aim for production purpose.

**Comment by guicho271828** [comment]
I think I have to implement [simpath](https://github.com/guicho271828/simpath) for dog-fooding this library.

**Comment by scymtym** [comment]
> Thanks for the report. Do you have plan to use this library like real soon? I would recommend CL-CUDD if you aim for production purpose.

No, this came up because it is one of the things blocking sionescu/fiveam#40.

**Comment by guicho271828** [comment]
no, go ahead! change it. It is my responsibility to fix my library. Those who cannot follow the latest changes should be left behind and dropped.

**Comment by guicho271828** [comment]
I mean in this case this library should be dropped from quicklisp.

**Comment by guicho271828** [comment]
well, I ve fixed it anyways...

**Comment by guicho271828** [comment]
Changing the library to break the other libs are completely OK, that is what quicklisp's deprecation policy is for. It does not mean the lib is never re-submitted either. Common Lisp is on the verge of extinction, and while being conservative may sound keeping it alive, in fact it is otherwise: causing the lack of advancement.

**Comment by scymtym** [comment]
Thanks.
